### DISCUSSION TITLE
How to disable javascript content-assist

### DISCUSSION THREAD

**Discussion by damianozucconi** [mickaelistria/eclipse-bluesky#1201] [Q&A]
I'm using version 1.0.5
The content assist in JavaScript is useless for my Sencha code and a little bit "invasive".
Can I disable it ?
![image](https://user-images.githubusercontent.com/12892164/236490117-03555459-4f8d-4c94-b20e-e2abb1dcf724.png)

**Comment by vrubezhny** [comment]
At the moment it's not possible to disable content assist.  
But maybe it's a question to https://github.com/typescript-language-server/typescript-language-server - I mean returning wrong content assist proposals for such a context?

**Reply by damianozucconi** [reply]
The content of assist maybe right. But if I need to add a character into a label or property name, the content assist popups instantly and I'm stucked into this window. If I want to move on the under row I need to press "ESC" and resume my edit.
![image](https://user-images.githubusercontent.com/12892164/236760781-f36042b9-f877-4ae6-bf11-627b0d0598f8.png)
It would be nice if I could activate it only with CTRL+SPACE

**Comment by mickaelistria** [comment]
I agree with @vrubezhny , the issue is more about the completion proposals not matching your needs and being useless in this context more than the actual behavior of the popup. So this should probably be reported first to typescript-language-server to make the content useful to you.

> It would be nice if I could activate it only with CTRL+SPACE

A contribution that allows to enable this behavior (ie set auto-activation delay to infinite and remove  that would be welcome, but it more a preference to add as part of Eclpse Platform Text framework. You'll find there
```java
setAutoActivationDelay(0);
enableColoredLabels(true);
enableAutoActivateCompletionOnType(true);
```
Those `true` constants could be replaced by preference settings.

**Comment by nelson777** [comment]
Actually it's not just the completition proposals. Now that AI assist is available it interferes with code suggestions from tools like Codeium or Copilot. I mean, don't get me wrong, sometimes we still need code-completition, but I would much rather seeing this disabled by default and only available when we press CTRL-SPACE.

Would it be OK if I opened an issue about this so it can be in the backlog as an enhancement ?
